src/wallet/wallet.h

```diff
@@ -986,4 +984,37 @@ class CAccount
     }
 };
 
+static CCriticalSection cs_wallets;
+class CWallets
+{
+private:
+    static std::vector<CWallet *> wallets;
+public:
+    static CWallet* defaultWallet()
+    {
+        LOCK(cs_wallets);
+        if (!wallets.size())
+            return NULL;
+        return wallets[0];
+    }
+    static void addWallet(CWallet *newWallet)
+    {
+        LOCK(cs_wallets);
+        wallets.push_back(newWallet);
+    }
+    static void flushAllWallets(bool shutdown)
+    {
+        LOCK(cs_wallets);
+        for(CWallet *pWallet : wallets)
+            pWallet->Flush(shutdown);
```

Comment:
Does cs_wallets need to be held for access to (flushing) the wallet itself, or only for the wallets vector? If the former, then defaultWallet will fail to guard it; if the latter, the lock here is unnecessarily long-held.